pkg/util/net.go

```diff
@@ -81,3 +85,42 @@ func CreateLocalhostListenerOnPort(port int) (tcpnet.Listener, error) {
 
 	return ln, nil
 }
+
+func AddToNoProxyEnv(additionalEntries ...string) error {
+	entries := map[string]struct{}{}
+
+	// put both the NO_PROXY and no_proxy elements in a map to avoid duplicates
+	addNoProxyEnvVarEntries(entries, "NO_PROXY")
+	addNoProxyEnvVarEntries(entries, "no_proxy")
+
+	for _, entry := range additionalEntries {
+		entries[entry] = struct{}{}
+	}
+
+	noProxyEnv := strings.Join(mapKeys(entries), ",")
+
+	// unset the lower-case one, and keep only upper-case
+	os.Unsetenv("no_proxy")
+	return errors.Wrap(os.Setenv("NO_PROXY", noProxyEnv), "error updating NO_PROXY")
+}
+
+func mapKeys(entries map[string]struct{}) []string {
+	keys := make([]string, 0, len(entries))
+	for k := range entries {
+		keys = append(keys, k)
+	}
+
+	// sort keys to avoid issues with map key ordering in go future versions on the unit-test side
+	sort.Strings(keys)
```

Comment:
I know the sort is unnecessary on production, but guarantees key ordering during unit testing, which is not a problem in current go, but future go versions will change this.